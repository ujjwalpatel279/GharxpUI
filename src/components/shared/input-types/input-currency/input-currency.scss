// ========================================================================
// COMPONENTS/SHARED/INPUT-TYPES/CURRENCY
// ========================================================================

@import '~sassHelpers';

.c-input-currency {
  position: relative;

  &__icon {
    position: absolute;
    top: 1px;
    bottom: 1px;
    left: 0;
    display: flex;
    align-items: center;
    padding-left: spacer(2);
    padding-right: spacer(2);
    border-right: 1px solid color(grey3);
  }

  &.isRequired .c-input-currency__icon {
    padding-left: spacer(3);
  }

  input {
    padding-left: spacer(10);
  }
}
