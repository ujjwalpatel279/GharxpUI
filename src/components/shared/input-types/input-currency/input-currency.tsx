import React, { forwardRef, KeyboardEvent } from 'react';
import { useForm } from 'react-hook-form';

import { Input } from '../';
import { Icon } from '../..';

import './input-currency.scss';

interface InputCurrencyProps {
  id: string;
  placeHolder?: string;
  required?: boolean;
  error?: boolean;
  disabled?: boolean;
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string;
  onKeyUp?: (e: KeyboardEvent<HTMLInputElement>) => void;
}

type Ref = HTMLInputElement;

export const InputCurrency = forwardRef<Ref, InputCurrencyProps>(
  ({ id, required, error, disabled, placeHolder, setValue, value, onKeyUp }, ref) => {
    let inputCurrencyClasses = 'c-input-currency';

    if (required) {
      inputCurrencyClasses += ' isRequired';
    }

    return (
      <div className={inputCurrencyClasses}>
        <span className="c-input-currency__icon">
          <Icon name="pound" />
        </span>
        <Input
          type="number"
          error={error}
          id={id}
          placeHolder={placeHolder}
          ref={ref}
          setValue={setValue}
          value={value}
          onKeyUp={onKeyUp}
          required={required}
          disabled={disabled}
        />
      </div>
    );
  },
);

InputCurrency.displayName = 'InputCurrency';
