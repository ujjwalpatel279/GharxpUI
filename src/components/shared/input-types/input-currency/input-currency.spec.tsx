import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { InputCurrency } from './input-currency';

describe('Input Currency Component', () => {
  afterEach(cleanup);

  it('renders the currency component', () => {
    const { container } = render(<InputCurrency id="amount" required={false} ref={() => null} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('aria-invalid is false', () => {
    const { queryByPlaceholderText } = render(
      <InputCurrency id="amount" error={false} placeHolder="testInput" required={false} ref={() => null} />,
    );

    expect(queryByPlaceholderText('testInput').getAttribute('aria-invalid')).toBe('false');
  });

  it('aria-invalid is true', () => {
    const { queryByPlaceholderText } = render(
      <InputCurrency id="amount" error={true} placeHolder="testInput" required={false} ref={() => null} />,
    );

    expect(queryByPlaceholderText('testInput').getAttribute('aria-invalid')).toBe('true');
  });

  it('input value is initially empty and has a value once a value is entered', () => {
    const { queryByPlaceholderText } = render(
      <InputCurrency id="amount" error={true} placeHolder="testInput" required={false} ref={() => null} />,
    );

    const input = queryByPlaceholderText('testInput') as HTMLInputElement;
    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: '1234' } });
    expect(input.value).toBe('1234');
  });
});
