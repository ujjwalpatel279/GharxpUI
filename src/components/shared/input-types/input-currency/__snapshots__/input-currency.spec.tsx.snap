// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Currency Component renders the currency component 1`] = `
<div
  class="c-input-currency"
>
  <span
    class="c-input-currency__icon"
  >
    <svg
      class="c-icon"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M8.972 1.111c1.178 0 2.295 0.26 3.349 0.781l-0.568 1.237c-0.955-0.478-1.876-0.716-2.763-0.716-0.763 0-1.34 0.192-1.73 0.577-0.385 0.385-0.577 1.011-0.577 1.879v2.744h3.926v1.182h-3.926v2.056c0 0.62-0.102 1.141-0.307 1.563-0.198 0.422-0.527 0.769-0.986 1.042h7.396v1.433h-9.573v-1.312c1.271-0.291 1.907-1.194 1.907-2.707v-2.075h-1.842v-1.182h1.842v-2.94c0-1.104 0.347-1.972 1.042-2.605 0.695-0.639 1.631-0.958 2.809-0.958z"
      />
    </svg>
  </span>
  <input
    aria-invalid="false"
    class="c-input"
    id="amount"
    name="amount"
    type="number"
  />
</div>
`;
